### PR TITLE
Fix retry_codes when empty

### LINKED ISSUE #71: Timeout parameter doesn't work when retry is configured
Originally reported by @frankyn

### What

Setting CallOptions timeout does nothing when the RPC is configured to retry.

### To reproduce

```
video_service_client = Google::Cloud::VideoIntelligence::V1beta1::VideoIntelligenceServiceClient.new
features             = [Google::Cloud::Videointelligence::V1beta1::Feature::LABEL_DETECTION]                                                                                           
options = Google::Gax::CallOptions.new(timeout: 60)
path = "gs://cloudmleap/video/next/volleyball_court.mp4"
                                                                                                                  
operation = video_service_client.annotate_video(path, features, options: options) do |op|
  ...
end
```

### Expected behavior

The call retries with an initial timeout of 60s.

### Observed behavior

The call fails with "INVALID_ARGUMENT" due to an initial timeout of 19s; the configured timeout is ignored. It is possible to change the timeout only by manually configuring the full backoff settings.

### PR DESCRIPTION
**Pull request by blowmage**
[fixes #71]

### CODE REVIEW

**Comment by jbolinger** [comment]
@blowmage do you know what's going on with the tests on 2.2?

**Comment by blowmage** [comment]
The latest version of protobuf dropped the pre-compiled file for 2.2

File: spec/google/gax/api_callable_spec.rb [hunk 1]
```diff
@@ -353,7 +353,7 @@ module Gax
       my_callable = Google::Gax.create_api_call(
         func, CallSettings.new(timeout: 0, retry_options: retry_options)
       )
-      expect { my_callable.call }.to raise_error(Google::Gax::RetryError)
+      expect { my_callable.call }.to raise_error(CustomException)
```
Comment (blowmage) [inline spec/google/gax/api_callable_spec.rb]: It is unfortunate that the test was specifying the wrong behavior.

**Comment by blowmage** [comment]
@jbolinger Once #159 is approved I'll merge and rebase this PR to get the fix. Then this PR should pass CI.

**Pull request merged**